src/module.ml

```diff
@@ -140,3 +153,26 @@ let dir t =
   Path.parent_exn file.path
 
 let set_pp t pp = { t with pp }
+
+let to_sexp { name; impl; intf; obj_name ; pp } =
+  let open Sexp.To_sexp in
+  record
+    [ "name", Name.to_sexp name
+    ; "obj_name", string obj_name
+    ; "impl", (option File.to_sexp) impl
+    ; "intf", (option File.to_sexp) intf
+    ; "pp", (option string) (Option.map ~f:(fun _ -> "has pp") pp)
+    ]
+
+let deprecate t =
+  { t with
+    intf = None
+  ; impl =
+      Some (
+        let impl = Option.value_exn t.impl in
```

Comment:
What happens if we use `(wrapped (transition ""))` with `(modules_without_implementation x)`?

Reply:
Good call. I fixed this and added a test case.